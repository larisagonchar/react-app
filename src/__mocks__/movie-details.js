import { GENRES_LIST } from '../constants/genre.constants';

export const movieDetailsMock = {
  genres: [GENRES_LIST[1].item, GENRES_LIST[2].item],
  release_date: 2004,
  title: 'Pulp Fiction',
  vote_average: 9.8,
  overview: 'Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are two hit men who are out to retrieve a suitcase stolen from their employer, mob boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to lose his fight. The lives of these seemingly unrelated people are woven together comprising of a series of funny, bizarre and uncalled-for incidents.â€”Soumitra',
  runtime: 130,
  poster_path: ''
};