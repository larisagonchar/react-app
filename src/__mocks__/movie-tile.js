import { GENRES_LIST } from '../constants/genre.constants';

export const movieTileMock = {
  genres: [GENRES_LIST[1].item, GENRES_LIST[2].item],
  release_date: 2004,
  title: 'Pulp Fiction',
  poster_path: ''
};