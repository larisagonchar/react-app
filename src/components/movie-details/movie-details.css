@import '../../styles/colors.css';
@import '../../styles/sizes.css';

.movie-details {
  display: flex;
  flex-direction: column;
  padding: var(--indentation-size-large);
  gap: var(--gap-size-large);
  background-color: var(--menu-background-color);
}

.movie-details__main {
  display: flex;
  gap: var(--gap-size-large);
}

.movie-details__title {
  font-size: var(--font-size-large);
  color: var(--white-color);
  font-weight: var(--font-weight);
  text-transform: uppercase;
}

.movie-details__rating {
  color: var(--white-color);
  border-radius: var(--border-radius-large);
  border: var(--border-size) solid var(--white-color);
  padding: var(--indentation-size-medium);
  font-size: var(--font-size-medium);
}

.movie-details__container {
  display: flex;
  align-items: center;
  gap: var(--gap-size-medium);
}

.movie-details__year, .movie-details__time {
  color: var(--primary-color);
  font-size: var(--font-size-medium);
  gap: var(--gap-size-large);
  margin: var(--indentation-size-large) 0;
  font-weight: var(--font-weight);
}

.movie-details__description {
  font-size: var(--font-size-medium);
  color: var(--white-color);
  opacity: 0.6;
  font-weight: var(--font-weight)/2;
}