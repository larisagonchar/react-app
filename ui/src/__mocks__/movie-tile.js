import { GENRES_LIST } from '../constants/genre.constants';

export const movieTileMock = {
  genres: [GENRES_LIST[1].item, GENRES_LIST[2].item],
  release_date: '2016-12-29',
  title: 'Pulp Fiction',
  poster_path: 'https://image.tmdb.org/t/p/w500/ylXCdC106IKiarftHkcacasaAcb.jpg'
};