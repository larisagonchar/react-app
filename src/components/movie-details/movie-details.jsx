import React from 'react';
import './movie-details.css';
import MovieGenre from './elements/movie-genre/movie-genre';
import MovieImage from './elements/movie-image/movie-image';
import Title from '../title/title';
import Header from '../header/header';
import { useLoaderData, useNavigate, useSearchParams } from 'react-router-dom';
import { mapMovieFromServiceToUIMovie } from 'src/mappers/movie-form.mapper';

const MovieDetails = () => {
  const selectedMovie = useLoaderData();
  const navigate = useNavigate();
  const [searchParams] = useSearchParams();

  const movie = mapMovieFromServiceToUIMovie(selectedMovie);

  const handleSearchClick = () => {
    navigate(`/?${searchParams}`);
  };

  return (
    <section className='movie-details'>
      <Header isSearchButtonVisible={true} onSearch={handleSearchClick}/>

      <div className='movie-details__main'>
        <MovieImage image={movie.poster_path}/>

        <div className='movie-details__content'>
          <div className='movie-details__container'>
            <Title title={movie.title}/>
            <span className='movie-details__rating'>{movie.vote_average}</span>
          </div>

          <MovieGenre genres={movie.genres}/>

          <div className='movie-details__container'>
            <span className='movie-details__year'>{movie.release_date}</span>
            <span className='movie-details__time'>{movie.runtime}</span>
          </div>

          <span className='movie-details__description'>{movie.overview}</span>
        </div>
      </div>
    </section>
  );
}

export default MovieDetails;