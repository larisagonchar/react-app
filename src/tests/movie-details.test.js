import { render } from '@testing-library/react';
import MovieDetails from 'components/movie-details/movie-details';
import { movieDetailsMock } from '../__mocks__/movie-details';

test('MovieDetails component should render movie details', () => {
  const { asFragment } = render(<MovieDetails selectedMovie={movieDetailsMock} />);

  expect(asFragment).toMatchSnapshot();
});