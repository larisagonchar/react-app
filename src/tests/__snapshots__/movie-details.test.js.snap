// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails component render movie details 1`] = `[Function]`;
